nselib/url.lua

```diff
@@ -177,8 +178,22 @@ function parse(url, default)
     parsed.params = p
     return ""
   end)
+
   -- path is whatever was left
   parsed.path = url
+
+  -- Checks for folder route and extension
+  if(string.match(parsed.path, "/$")) then
+    parsed.is_folder = true
+  else
+    parsed.is_folder = false
+    local split_str = stdnse.strsplit('%.', parsed.path)
+    if(split_str and #split_str > 1) then
+      parsed.extension = stdnse.strsplit("%;", split_str[#split_str])[1]
+    end
```

Comment:
I think that all we need to do is to prohibit dot in the capture group, so:
```lua
parsed.extension = parsed.path:match("%.([^/.;]+)%f[;\0][^/]*$")
```

This is the effect:
```
$ lua
Lua 5.3.4  Copyright (C) 1994-2017 Lua.org, PUC-Rio
> test={"/files/tmp.php/lastfile.txt",
>> "/files/tmp.php/lastfile",
>> "/files.",
>> "https://svn.nmap.org/nmap-releases/nmap-7.60/CHANGELOG",
>> "foo/.",
>> "http://php.net/manual/en/function.pathinfo.php"}
> for _,s in ipairs(test) do print(s,s:match("%.([^/.;]+)%f[;\0][^/]*$")) end
/files/tmp.php/lastfile.txt     txt
/files/tmp.php/lastfile nil
/files. nil
https://svn.nmap.org/nmap-releases/nmap-7.60/CHANGELOG  nil
foo/.   nil
http://php.net/manual/en/function.pathinfo.php  php
```